### PR TITLE
FullSignReconstruction moved from airgapped to hotnode #LSC-193 #LSC-192

### PR DESCRIPTION
**Pull request by swelf19**
Key steps to move reconstruction logic from airgapped to hot node.
1) 1) In case of ordinary DKG generation, at the end of DKG procedure we send BLS public key as part of response to HotNode https://github.com/lidofinance/dc4bc/blob/refactor/moving-signature-reconstruction/airgapped/dkg.go#L352
    2) In case of ReinitDKG we send BLS public key as well but in a different place https://github.com/lidofinance/dc4bc/blob/refactor/moving-signature-reconstruction/airgapped/dkg.go#L61

2) When hotnode have partialsigns enough to reconstruct signature, signature reconstruction starts immediately https://github.com/lidofinance/dc4bc/blob/refactor/moving-signature-reconstruction/fsm/state_machines/signing_proposal_fsm/actions.go#L255.
After signature reconstructed, hotnode broadcasts it to all participants

state_signing_* operations are not stored anymore in operation log, cause they do mean nothing to airgapped state

### CODE REVIEW

File: client/services/node/node_service.go [hunk 1]
```diff
@@ -271,6 +272,23 @@ func (s *BaseNodeService) executeOperation(operation *types.Operation) error {
 		if err := s.storage.Send(operation.ResultMsgs...); err != nil {
 			return fmt.Errorf("failed to post messages: %w", err)
 		}
+	} else {
+		//for now only ReinitDKG can has the OperationProcessed event
```
Comment (callmepak) [inline client/services/node/node_service.go:276]: I think, that separate "if" is better than just "else"

**Review by pr0n00gler — APPROVED** [review]

**Review by sotnikov-s — APPROVED** [review]

File: airgapped/airgapped.go [hunk 2]
```diff
@@ -179,18 +151,12 @@ func (am *Machine) ProcessOperation(operation client.Operation, storeOperation b
 			operation.ID, err)
 	}
 
-	if storeOperation {
+	if storeOperation && !strings.HasPrefix(string(operation.Type), "state_signing_") {
```
Comment (oopcode) [inline airgapped/airgapped.go:154]: There is at least two places where the same cumbersome check is performed. What do you think about creating an Operation.IsSigning operation method?

File: client/flow_test.go [hunk 3]
```diff
@@ -776,6 +783,19 @@ func convertDKGMessageto0_1_4(orig types.ReDKG) types.ReDKG {
 		if fsm.Event(m.Event) == dkg_proposal_fsm.EventDKGDealConfirmationReceived && m.SenderAddr == m.RecipientAddr {
 			continue
 		}
+		if fsm.Event(m.Event) == dkg_proposal_fsm.EventDKGMasterKeyConfirmationReceived {
+			// removing PolyPob from old log
```
Comment (oopcode) [inline client/flow_test.go:787]: PolyPob -> PolyPub

File: client/services/node/node_service.go (hunk 1 above)
Comment (oopcode) [inline client/services/node/node_service.go:276]: remove "can"

**Review by oopcode — COMMENTED** [review]
i'm not sure that moving the reconstruction logic to the FSM internals is a nice idea, seems like our FSM doesn't do any cryptography at all besides that bit

**Review by oopcode — APPROVED** [review]

**Review by oopcode — APPROVED** [review]

**Pull request merged**